param/__init__.py

```diff
@@ -775,22 +773,37 @@ def __init__(self,default=(0,0),length=None,**params):
             self.length = length
 
         self._check(default)
-        Parameter.__init__(self,default=default,**params)
+        super(Tuple,self).__init__(default=default,**params)
+
 
     def _check(self,val):
+        if val is None and self.allow_None:
+            return
+
         if not isinstance(val,tuple):
             raise ValueError("NumericTuple '%s' only takes a tuple value."%self._attrib_name)
 
         if not len(val)==self.length:
             raise ValueError("%s: tuple is not of the correct length (%d instead of %d)." %
                              (self._attrib_name,len(val),self.length))
-        for n in val:
-            if not _is_number(n):
-                raise ValueError("%s: tuple element is not numeric: %s." % (self._attrib_name,str(n)))
+
 
     def __set__(self,obj,val):
         self._check(val)
-        super(NumericTuple,self).__set__(obj,val)
+        super(Tuple,self).__set__(obj,val)
+
+
+
+class NumericTuple(Tuple):
+    """A numeric tuple Parameter (e.g. (4.5,7.6,3)) with a fixed tuple length."""
+
+    __slots__ = []
```

Comment:
Note that we add `__slots__` if Parameter subclass authors don't do it themselves:

```python
class ParameterMetaclass(type):
    [...]
    def __new__:
        [...]
        # To get the benefit of slots, subclasses must themselves define
        # __slots__, whether or not they define attributes not present in
        # the base Parameter class.  That's because a subclass will have
        # a __dict__ unless it also defines __slots__.
        if '__slots__' not in classdict:
            classdict['__slots__']=[]
```
https://github.com/ioam/param/blob/master/param/parameterized.py#L206

That seemed like a good idea at the time (when nobody had ever heard of `__slots__`, and when we had to discover things like this ourselves/read python's source code).